### PR TITLE
List View: Design updates

### PR DESCRIPTION
**Pull request by shaunandrews**
WIP

image

Still a lot to do, but this PR starts by
- reducing the row height down (48px to 36px), and overall spacing.
- adding the more (...) menu everywhere.
- removing the descender lines.
- arranging movers side-by-side.
- adding a hover affect.

There's still a lot of work to be done. Here's how the changes look in the experimental Navigation screen, which shows the movers:

image

### CODE REVIEW

**Comment by draganescu** [comment]
Why is it a draft? It contains more than enough changes for an iteration 💯

**Comment by shaunandrews** [comment]
image

I pushed some changes:
- Made the popover wider
- Forced nested blocks to always show
- Forced the movers to show
- Switched away from `table` elements to basic `div`'s to have better control over focus and general styling
- Added an alternative for the nested descender indicator, this time its a more subtle series of dots

> Why is it a draft? It contains more than enough changes for an iteration

I'm sure it does. For now, this is kind of a sandbox to see how things feel in practice. It might be mergeable, or maybe we end up splitting it up.

**Comment by SchneiderSam** [comment]
Sooo cool. Big Game Changer....

I was about to test it with http://gutenberg.run/24419, but unfortunately it does not work. I'm getting an error. I've tried it several times... @aduth

**Comment by enriquesanchez** [comment]
This looks great and works as expected. I didn't notice any loss in functionality due to the use of `divs`. 

One thing I came across is we'd probably want to truncate or do something about long strings of text:

Screen Shot 2020-08-13 at 16 00 11

File: packages/block-editor/src/components/block-navigation/style.scss [hunk 1]
```diff
@@ -1,244 +1,121 @@
...
+	border-left: 2px solid $gray-700;
+	margin: 0 6px 0;
 }
 
+// Hide block number and level description. (i.e. Block 2 of
+// 3, Level 2). This is mostly useful for screen readers.
 .block-editor-block-navigation-block-slot__description,
 .block-editor-block-navigation-block-select-button__description,
 .block-editor-block-navigation-appender__description {
 	display: none;
```
Comment (tellthemachines) [inline packages/block-editor/src/components/block-navigation/style.scss:120]: We can't use `display: none` if the content is meant to be screen reader accessible. 
Current best practice in this repo is to wrap the screen-reader-only component in [VisuallyHidden](https://github.com/WordPress/gutenberg/blob/abbdb2bfae951bed85c568ece12620bacd1e483f/packages/components/src/visually-hidden/README.md#L1). This ensures it'll have the appropriate styling.

File: packages/components/src/tree-grid/index.js [hunk 2]
```diff
@@ -150,10 +150,10 @@ export default function TreeGrid( { children, ...props } ) {
 	return (
 		<RovingTabIndexContainer>
 			{ /* Disable reason: A treegrid is implemented using a table element. */ }
```
Comment (tellthemachines) [inline packages/components/src/tree-grid/index.js:152]: Need to update the disable reason here.

**Review by tellthemachines — CHANGES_REQUESTED** [review]
This is working pretty well! Just a few comments below.

Also, if we're showing the "More options" menu (I'm not fully convinced we should 😅 ), we'll probably need to hide some of the options. E.g. trying to use "Move to" is awkward from the modal and impossible from the top toolbar navigator.
Screen Shot 2020-08-14 at 5 18 47 pm

Finally:

* Lint checks are failing: run `npm run lint-js` once you've finished making changes and fix the resulting errors.
* Treegrid snapshots need updating: run `npm run test-unit -- -u` to update them.

**Comment by paaljoachim** [comment]
Here is the PR for **WIP: Add drag and drop to List View**
https://github.com/WordPress/gutenberg/pull/23952

**Comment by paaljoachim** [comment]
Can we get an update to how things are going here?
Thanks!

**Comment by paaljoachim** [comment]
I was finally able to test this out by not using gutenberg.run.

I followed this doc: https://developer.wordpress.org/block-editor/tutorials/devenv/
(Which we added to the core editor yesterday.)

and I am following steps in how to test run a PR. I will later today add an issue and then create a PR for it.

Here is an animated gif. Showing the Navigation screen structure.
Nav-structure

What is interesting is the clarity in when an element is selected. But the black background color creates a too strong contrast. The black could perhaps be a greyish color instead. As it might just need a much more subtitle background color. I am not sure about the dots. Having the old tree diagram lines seemed easier to follow.
All in all I am not sure about the direction seen in this PR in relation to what I experienced in the Nav screen structure.

**Comment by noahtallen** [comment]
@shaunandrews is there anything specific we can do to help move this forward?

**Comment by shaunandrews** [comment]
> is there anything specific we can do to help move this forward?

I likely tried doing _too much_ in this PR. I think we could probably break some of this up into smaller branches, like:

- reducing the row height down (48px to 36px), and overall spacing.
- removing (and potentially replacing) the descender lines.
- arranging movers side-by-side.
- adding a hover affect.

**Comment by shaunandrews** [comment]
Cherry-picked a few things from this bigger PR into a smaller one over in #28029

**Comment by draganescu** [comment]
@shaunandrews as the new screen has no list view anymore, is this design update maybe outside of the Navigation editor scope?

**Comment by shaunandrews** [comment]
I think we'll bring the list view back to the navigation editor, so it still makes sense to improve it as part of that effort.

**Comment by getdave** [comment]
@draganescu @shaunandrews Should we remove this from `PRs pending review`?

**Pull request closed without merging**